addons/mail/static/src/messaging/entity/thread/thread.js

```diff
@@ -0,0 +1,674 @@
+         */
+        static get allOrderedAndPinnedChannels() {
+            return this.allChannels.sort((c1, c2) => c1.displayName < c2.displayName ? -1 : 1);
+        }
+
+        /**
+         * @returns {mail.messaging.entity.Thread[]}
+         */
+        static get allOrderedAndPinnedChats() {
+            return this.allPinnedChannels
+                .filter(channel => channel.channel_type === 'chat')
+                .sort((c1, c2) => c1.displayName < c2.displayName ? -1 : 1);
+        }
+
+        /**
+         * @returns {mail.messaging.entity.Thread[]}
+         */
+        static get allOrderedAndPinnedMultiUserChannels() {
+            return this.allPinnedChannels
+                .filter(channel => channel.channel_type === 'channel')
+                .sort((c1, c2) => c1.displayName < c2.displayName ? -1 : 1);
+        }
+
+        /**
+         * @returns {mail.messaging.entity.Thread[]}
+         */
+        static get allOrderedAndPinnedMailboxes() {
+            return this.allMailboxes
+                .filter(mailbox => mailbox.isPinned)
+                .sort((mailbox1, mailbox2) => {
+                    if (mailbox1.id === 'inbox') {
+                        return -1;
+                    }
+                    if (mailbox2.id === 'inbox') {
+                        return 1;
+                    }
+                    if (mailbox1.id === 'starred') {
+                        return -1;
+                    }
+                    if (mailbox2.id === 'starred') {
+                        return 1;
+                    }
+                    const mailbox1Name = mailbox1.displayName;
+                    const mailbox2Name = mailbox2.displayName;
+                    mailbox1Name < mailbox2Name ? -1 : 1;
+                });
+        }
+
+        /**
+         * @returns {mail.messaging.entity.Thread[]}
+         */
+        static get allPinnedChannels() {
+            return this.allChannels.filter(channel => channel.isPinned);
+        }
+
+        /**
+         * @returns {mail.messaging.entity.Thread[]}
+         */
+        static get allUnreadChannels() {
+            return this.allChannels.filter(
+                channel => channel.message_unread_counter > 0
+            );
+        }
+
+        /**
+         * @param {integer} id
+         * @returns {mail.messaging.entity.Thread|undefined}
+         */
+        static channelFromId(id) {
+            return this.allChannels.find(channel => channel.id === id);
+        }
+
+        /**
+         * Create a channel, which is a special kind of thread on model
+         * 'mail.channel' with multiple members.
+         *
+         * @param {Object} param0
+         * @param {boolean} [param0.autoselect=false]
+         * @param {string} [param0.autoselectChatWindowMode]
+         * @param {string} param0.name
+         * @param {integer|undefined} [param0.partnerId=undefined]
+         * @param {string|undefined} [param0.public=undefined]
```

Comment:
isn't it obvious that optional params take undefined by default? Never seen it documented that way.